File: solutions/851.goat-latin/goat-latin.cpp
class Solution {
public:
    string toGoatLatin(string S) {
        stringstream iss(S), oss;
        string vowels("aeiouAEIOU"), a, word;
        while(iss >> word) {
            a.push_back('a');
            if(vowels.find_first_of(word[0]) != string :: npos)
                oss << ' ' << word << "ma" << a;
            else oss << ' ' << word.substr(1) << word[0] << "ma" << a;
        }
        return oss.str().substr(1);
    }
};

File: solutions/851.goat-latin/goat-latin.java
class Solution {
    public String toGoatLatin(String S) {
        Set<Character> vowels = new HashSet<>();
        for(char c : "aieouAIEOU".toCharArray()) vowels.add(c);
        StringBuilder res = new StringBuilder();
        StringBuilder aStr = new StringBuilder();
        for(String word : S.split(" ")) {
            aStr.append('a');
            res.append(" ");
            if(vowels.contains(word.charAt(0))) res.append(word);
            else res.append(word.substring(1)).append(word.charAt(0));
            res.append("ma").append(aStr);
        }
        return res.toString().substring(1);
    }
}